lib/net.js

```diff
@@ -121,6 +121,9 @@ function Socket(options) {
   this._hadError = false;
   this._handle = null;
   this._host = null;
+  this._unref = false;
+  this._setNoDelay = null;
+  this._setKeepAlive = null;
```

Comment:
Similarly, are there better defaults for this? It might be worth splitting up into two flags instead of always using an array.